tracing-error/src/backtrace.rs

```diff
@@ -170,8 +169,46 @@ impl fmt::Display for SpanTrace {
     }
 }
 
+struct SpanTraceFrame<'a> {
+    metadata: &'a Metadata<'a>,
+    fields: &'a str,
+}
+
+impl<'a> fmt::Debug for SpanTraceFrame<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "{{ target: {:?}, name: {:?}",
+            self.metadata.target(),
+            self.metadata.name()
+        )?;
+
+        if !self.fields.is_empty() {
+            write!(f, ", fields: {:?}", self.fields)?;
+        }
+
+        if let Some((file, line)) = self
+            .metadata
+            .file()
+            .and_then(|file| self.metadata.line().map(|line| (file, line)))
+        {
+            write!(f, ", file: {:?}, line: {:?}", file, line)?;
+        }
+
+        write!(f, " }}")?;
```

Comment:
TIOLI: i think the same formatting could possibly be accomplished in a more concise way using [`Formatter::debug_struct`](https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.debug_struct). not a blocker though!

Reply:
True, but that would result in extra newlines for alt format where the alt debug format would look like

```
SpanTrace [
    { 
        target: "custom_error", name: "do_another_thing", fields: "answer=42 will_succeed=false", file: "examples/examples/custom_error.rs", line: 43
    },
    { 
        target: "custom_error", name: "do_something", fields: "foo=\"hello world\"", file: "examples/examples/custom_error.rs", line: 38
    },
]
```

instead of fitting each frames output on one line. I initially did the Backtrace pr with debug_struct but they opted for a manual write version like this

Reply:
ah, gotcha. let's go with this, then, if it's closer to `std::backtrace`!